sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -899,21 +899,15 @@ object SparkSession {
 
         // No active nor global default session. Create a new one.
         val sparkContext = userSuppliedContext.getOrElse {
+          val sparkConf = new SparkConf()
+          SparkContext.getOrCreate(sparkConf)
+        }
+
+        options.foreach { case (k, v) => sparkContext.conf.set(k, v) }
+        if (!sparkContext.conf.contains("spark.app.name")) {
           // set app name if not given
           val randomAppName = java.util.UUID.randomUUID().toString
-          val sparkConf = new SparkConf()
-          options.foreach { case (k, v) => sparkConf.set(k, v) }
-          if (!sparkConf.contains("spark.app.name")) {
-            sparkConf.setAppName(randomAppName)
-          }
-          val sc = SparkContext.getOrCreate(sparkConf)
-          // maybe this is an existing SparkContext, update its SparkConf which maybe used
-          // by SparkSession
-          options.foreach { case (k, v) => sc.conf.set(k, v) }
```

Comment:
previously we set the given `options` to the newly created `SparkConf`, then create `SparkContext`, then set the `options` to `sc.conf` again, in case the `SparkContext.getOrCreate` returns an existing one. We can just create `SparkConf` and use it to create `SparkContext`, and then set `options` to `sc.conf`, so that we only need to set the conf once.